config/routes.rb

```diff
@@ -0,0 +1,18 @@
+Rails.application.routes.draw do
+  # verb "path", to: "controller#action"
+  get "/users/current", to: "users#current", as: "current_user"
+  resources :works
+  resources :users
+  get "/login", to: "users#login_form", as: "login"
+  post "/login", to: "users#login"
+  post "/logout", to: "users#logout", as: "logout"
+  #allow user to visit this page of they logged in
+  root to:'works#top'
+  post "/users/vote", to: "users#vote", as: "vote"
```

Comment:
I'd suggest making this a nested route under works.  Then you can have a votes controller with just a create method. 

```ruby
resources :works do
  resources :votes, only: [:create]
end